### PR TITLE
Fix TypeConverter tests on UAP

### PR DESCRIPTION
**Pull request by krwq**
Fixes: https://github.com/dotnet/corefx/issues/21330

Background: Tests were flaky because TypeConverter (product) is temporarily changing UI culture which is making tests fail as they expect `CultureInfo.CurrentCulture != CultureInfo.InvariantCulture`

This includes two fixes, each of them separately fixes the issue:
- product fix to not do that
- tests use RemoteExecutor and override culture from within

I recommend looking at this diff with `&w=1` / `?w=1`:
https://github.com/dotnet/corefx/commit/a246a010a20f4ecd70585527b7f15f8421fab724?w=1

cc: @tarekgh - Thank you a lot for help on this!

### CODE REVIEW

File: src/System.ComponentModel.TypeConverter/tests/DateTimeConverterTests.cs [hunk 1]
```diff
@@ -48,20 +48,25 @@ public static void ConvertFrom_WithContext_Negative()
         [Fact]
         public static void ConvertTo_WithContext()
         {
-            DateTimeFormatInfo formatInfo = (DateTimeFormatInfo)CultureInfo.CurrentCulture.GetFormat(typeof(DateTimeFormatInfo));
-            string formatWithTime = formatInfo.ShortDatePattern + " " + formatInfo.ShortTimePattern;
-            string format = formatInfo.ShortDatePattern;
-            DateTime testDateAndTime = new DateTime(1998, 12, 5, 22, 30, 30);
+            RemoteInvoke(() => {
+                CultureInfo.CurrentCulture = new CultureInfo("pl-PL");
+                DateTimeFormatInfo formatInfo = (DateTimeFormatInfo)CultureInfo.CurrentCulture.GetFormat(typeof(DateTimeFormatInfo));
```
Comment (tarekgh) [inline src/System.ComponentModel.TypeConverter/tests/DateTimeConverterTests.cs]: I know why you chose pl-PL :-)

**Comment by tarekgh** [comment]
LGTM.

File: src/System.ComponentModel.TypeConverter/tests/DateTimeConverterTests.cs (hunk 1 above)
Comment (krwq) [inline src/System.ComponentModel.TypeConverter/tests/DateTimeConverterTests.cs]: at random? 😉

Comment (krwq) [inline src/System.ComponentModel.TypeConverter/tests/DateTimeConverterTests.cs]: https://imgs.xkcd.com/comics/random_number.png

**Comment by krwq** [comment]
Please do not squash - changes are independent

**Comment by krwq** [comment]
I don't believe failure is related. @tarekgh you might want to take a look:
https://mc.dot.net/#/user/krwq/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/477a367e907e2f7d7e9474336280440763b0adb5/workItem/System.Globalization.Extensions.Tests/analysis/xunit/System.Globalization.Tests.IdnMappingIdnaConformanceTests~2FGetAscii_Invalid

**Comment by krwq** [comment]
@dotnet-bot test Windows x64 Debug Build please

File: src/System.ComponentModel.TypeConverter/tests/TypeConverterTests.cs [hunk 2]
```diff
@@ -138,18 +145,29 @@ public static void ProtectedMethods()
 
         private void RunProtectedMethods()
         {
-            Assert.Throws<NotSupportedException>(() => GetConvertFromException(null));
-            Assert.Throws<NotSupportedException>(() => GetConvertFromException("1"));
-            Assert.Throws<NotSupportedException>(() => GetConvertFromException(new BaseClass()));
-            Assert.Throws<NotSupportedException>(() => GetConvertToException(null, typeof(int)));
-            Assert.Throws<NotSupportedException>(() => GetConvertToException("1", typeof(int)));
-            Assert.Throws<NotSupportedException>(() => GetConvertToException(new BaseClass(), typeof(BaseClass)));
+            var tc = new TypeConverterHelper();
```
Comment (danmosemsft) [inline src/System.ComponentModel.TypeConverter/tests/TypeConverterTests.cs]: What is the purpose of introducing this class?

**Comment by danmosemsft** [comment]
There seem to be many other tests in other libraries that set CurrentCulture without using RemoteExecutorTestBase. We should fix those separately. Could you open an issue?

There are several other failures in the UAP run where seemingly the culture is wrong. I wish I could blame it on this but I suspect they are happening too consistently and they have some other cause.

**Comment by tarekgh** [comment]
> There are several other failures in the UAP run where seemingly the culture is wrong. I wish I could blame it on this but I suspect they are happening too consistently and they have some other cause.

Note that cultures in UWP apps is working differently than the non UWP. so the tests could be written not considering that. I didn't look though to tell what is the problem exactly.

**Comment by danmosemsft** [comment]
nano issue is https://github.com/dotnet/corefx/issues/21332

**Comment by danmosemsft** [comment]
Nano issue fixed in https://github.com/dotnet/corefx/pull/21264/commits/07c9fadf7296884790e9198a667db2fc8e7f1050

File: src/System.ComponentModel.TypeConverter/tests/TypeConverterTests.cs (hunk 2 above)
Comment (krwq) [inline src/System.ComponentModel.TypeConverter/tests/TypeConverterTests.cs]: no multi inheritance

**Pull request merged**